pkg/store/consolidationstore/store.go

```diff
@@ -0,0 +1,150 @@
+package consolidationstore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"github.com/storacha/go-ucanto/core/car"
+	"github.com/storacha/go-ucanto/core/dag/blockstore"
+	"github.com/storacha/go-ucanto/core/invocation"
+)
+
+const (
+	trackInvocationsPrefix   = "track/"
+	consolidateInvCIDsPrefix = "consolidate/"
+)
+
+// Store stores egress/track invocations and their corresponding
+// consolidate invocation CIDs, indexed by batch CID.
+// When a batch is tracked via space/egress/track, the receipt contains an effect
+// with a space/egress/consolidate invocation. We store both the track invocation
+// and the consolidate invocation CID for later retrieval.
+type Store interface {
+	// Put stores an egress/track invocation and consolidate invocation CID indexed by batch CID.
+	Put(ctx context.Context, batchCID cid.Cid, trackInv invocation.Invocation, consolidateInvCID cid.Cid) error
+
+	// GetTrackInvocation retrieves the egress/track invocation for a given batch CID.
+	GetTrackInvocation(ctx context.Context, batchCID cid.Cid) (invocation.Invocation, error)
```

Comment:
I believe this method is unused, are there plans for it in the future?

Reply:
yes, the idea is that the original track invocation will be needed to confirm the validity of the consolidate receipt for proper accounting in a trustless manner.